rostime/include/ros/time.h

```diff
@@ -151,7 +151,10 @@ namespace ros
 
     double toSec()  const { return (double)sec + 1e-9*(double)nsec; };
     T& fromSec(double t) {
-      sec = (uint32_t)floor(t);
+      int64_t sec64 = (int64_t)floor(t);
+      if (sec64 < 0 || sec64 > UINT_MAX)
+        throw std::runtime_error("Duration is out of dual 32-bit range");
```

Comment:
Thank you for pointing it out. Fixed in f824e46990f20c066596e614b7344ed9d03d87c8.